.github/workflows/check-changelog.yml

```diff
@@ -0,0 +1,35 @@
+#copied configuration from scikit-learn
+
+name: Check Changelog
+# This check makes sure that the changelog is properly updated
+# when a PR introduces a change in a test file.
+# To bypass this check, label the PR with "No Changelog Needed".
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+
+jobs:
+  check:
+    name: A reviewer will let you know if it is required or can be bypassed
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
+      - name: Check if tests have changed
+        id: tests_changed
+        run: |
+          set -xe
+          changed_files=$(git diff --name-only origin/main)
+          # Changelog should be updated only if tests have been modified
+          if [[ "$changed_files" =~ tests ]]
+          then
+            echo "check_changelog=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Check changelog entry
+        if: steps.tests_changed.outputs.check_changelog == 'true'
+        uses: scientific-python/action-towncrier-changelog@v1
```

Comment:
this doesn't work for us cause we don't use `towncrier`, which requires a separate file for each changelog entry. You can check scikit-learn's bot from before `towncrier` was a thing. If you look at the `1.4` branch, you'll find it.

Reply:
oooh got it, I will make it work with the older config then. I thought it will all fit together somehow. I will check it out and try to make it work the old way then.

Reply:
(hopefully) fixed in https://github.com/fairlearn/fairlearn/pull/1457/commits/263cd8dd1f3f1df0c5848374f3d659ad6ee1e7fd. could you check it out again? I see a doctest failure not related to my PR, I will inspect now.

Reply:
fixed the doctest, it was a `numpy` version related error interestingly. I was behind a version